Add hack/verify-test-code.sh

oomichi opened the pull request:
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This script checks coding style for test code.
It is useful to enforce developers to use framework methods for
keeping readable test code for example.

This new script takes 1 sec to verify all e2e test code.

Fixes: https://github.com/kubernetes/kubernetes/issues/77103

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

oomichi commented:
/retest

oomichi commented:
/test pull-kubernetes-verify

atoato88 commented:
Why do you include a commit  b2c6dc57 “Use ExpectNoError() for scalability and scheduling” in this PR?

Commit:
Add hack/verify-test-code.sh

This script checks coding style for test code.
It is useful to enforce developers to use framework methods for
keeping readable test code for example.

oomichi commented:
@atoato88 Thanks for pointing it out. That was my mess, I am fixing that.

oomichi commented:
/retest

oomichi commented:
/test pull-kubernetes-verify

oomichi commented:
/test pull-kubernetes-verify

oomichi commented:
/test pull-kubernetes-verify

oomichi commented:
/retest

oomichi commented:
/retest

oomichi commented:
/cc @spiffxp @fejta 

Could you take a look at this?
I'd like to avoid pointing "Please use ExpectNoError() instead" in e2e test reviewing process if possible.

fejta reviewed (approved):
/approve
/lgtm
very cool, thanks!

oomichi commented:
@fejta Thanks for your support ! :-)

oomichi commented:
/retest

oomichi commented:
/retest

oomichi commented:
/test pull-kubernetes-integration
